gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java

```diff
@@ -1840,6 +1911,34 @@ protected GenerateDataTemplateTask configureDataTemplateGeneration(Project proje
     return generateDataTemplatesTask;
   }
 
+  private String mapSourceSetToFeatureName(SourceSet sourceSet) {
+    String featureName = "";
+    switch (sourceSet.getName()) {
```

Comment:
This appears to be tech debt, since we have to maintain static mappings from what the "natural" configuration names are to what we expect them to be.
1. Is my understanding correct?
2. If so, would there be a benefit in getting users to migrate their usage (e.g. from `restClient` to `mainGeneratedRest`)? Ignoring the cost of migration, of course... I want to understanding if this is "hacky" or totally acceptable in the Gradle world.

Reply:
This... is hard to explain. The extra complexity is related to the legacy dataModel/dataTemplate/dataTemplate configurations and preserving compatibility.

The first step is to please read and understand Gradle's concept of a "feature variant". This [doc](https://docs.gradle.org/6.1/userguide/feature_variants.html) is crucial. The pegasus data-template jar is a perfect example of a feature variant. In my own words, it's an "alternate" artifact addressable at the same coordinates as the "primary" artifact, and having a parallel set of its own dependencies.

Now, when registering a "feature variant" Gradle generates its own configurations which can represent the same concepts as dataModel/dataTemplate. The key difference is that these are derived from the sourceSet containing the "feature", whereas legacy pegasus does much extra work to associate the `mainGeneratedDataTemplate` sourceset to `dataTemplate` and `dataModel` configurations.

When doing it the Gradle way, a sourceset called `mainGeneratedDataTemplate` will spawn configurations named `mainGeneratedDataTemplateApiElements` and `mainGeneratedDataTemplateRuntimeElements`.In theory, these are what consumers should be modifying instead

Long-term, my design recommendation is to rename the `mainGeneratedDataTemplate` SourceSet to simply `dataTemplate`. This will spawn `dataTemplateApi` and `dataTemplateImplementation` configurations to the project applying the pegasus plugin, and consumers can decide how they want to manipulate the dependencies.

Reply:
Got it, thanks for the explanation. I should document this somewhere 😃